### PR TITLE
NEW : add constant ORDER_REOPEN_TO_DRAFT

### PR DESCRIPTION
**Pull request by altairis-tof**
add new constant to allow to reopen an order to draft status instead of validated.

### CODE REVIEW

**Comment by dpriskorn** [comment]
LGTM this saves an extra click which I like 😃

**Comment by altairis-tof** [comment]
EDITED : my previous comment was not ok or became obsolete after another answer update.

**Comment by eldy** [comment]
The only people that has github permission to merge request are the 3 people in the group Yoda: so me, simnandez and atm-maxime.

**Comment by altairis-tof** [comment]
@eldy i was talking about thie PR https://github.com/Dolibarr/dolibarr/pull/18327#issuecomment-993888032 that was closed by Dennis

File: htdocs/commande/class/commande.class.php [hunk 1]
```diff
@@ -696,8 +696,10 @@ public function set_reopen($user)
 
 		$this->db->begin();
 
+		$newstatus = empty($conf->global->ORDER_REOPEN_TO_DRAFT) ? self::STATUS_VALIDATED : self::STATUS_DRAFT;
```
Comment (eldy) [inline htdocs/commande/class/commande.class.php:699]: To switch the status of an object (so an order) to the draft status, there is already a dedicated method called setDraft().
So you shoul modify the caller calling set_reopen to also call setDraft() if ORDER_REOPEN_TO_DRAFT is set).

Comment (altairis-tof) [reply htdocs/commande/class/commande.class.php:699]: ok i will do it in next days; thanks

**Comment by BebZ** [comment]
Why a "constant" and not a "permission" ? 

This constant will disallow to put validated / in progress orders to draft again ?

**Comment by eldy** [comment]
Still error in travis.

**Comment by altairis-tof** [comment]
ok i will ask @altairis-benjamin to finish the job.

**Comment by eldy** [comment]
Closed as opened since a too long time. Thanks to make another PR if necessary

**Pull request closed without merging**